sunkit_instruments/suvi/suvi.py

```diff
@@ -0,0 +1,820 @@
+    L1b_matches = ['-L1b-Fe093', '-L1b-Fe131', '-L1b-Fe171', '-L1b-Fe195', '-L1b-Fe284', '-L1b-He303']
+    if any(fn in os.path.basename(files[0]) for fn in composite_matches):
+        composites = True
+    elif any(fn in os.path.basename(files[0]) for fn in L1b_matches):
+        composites = False
+    else:
+        raise ValueError("First file "+files[0]+" does not look like a SUVI L1b file or L2 HDR composite.")
+
+    datas = []
+    headers = []
+    for tmp_file in files:
+        if verbose:
+            print('Reading', tmp_file)
+        # Test for L1b or composite based on the filename
+        if composites:
+            if any(fn in os.path.basename(tmp_file) for fn in composite_matches):
+                tmp_header, tmp_data = sunkit_instruments.suvi.read_suvi(tmp_file)
+                datas.append(tmp_data)
+                headers.append(tmp_header)
+            else:
+                warn_user("File "+tmp_file+" does not look like a SUVI L2 HDR composite. Skipping.")
+        else:
+            if any(fn in os.path.basename(tmp_file) for fn in L1b_matches):
+                if despike_L1b:
+                    tmp_header, tmp_data, dqf_mask = sunkit_instruments.suvi.read_suvi(tmp_file, return_DQF=True)
+                    tmp_data = despike_L1b_image((tmp_data, dqf_mask))
+                else:
+                    tmp_header, tmp_data = sunkit_instruments.suvi.read_suvi(tmp_file)
+                if only_long_exposures:
+                    if 'long_exposure' in tmp_header['SCI_OBJ']:
+                        datas.append(tmp_data)
+                        headers.append(tmp_header)
+                elif only_short_exposures:
+                    if 'short_exposure' in tmp_header['SCI_OBJ']:
+                        datas.append(tmp_data)
+                        headers.append(tmp_header)
+                elif only_short_flare_exposures:
+                    if 'short_flare_exposure' in tmp_header['SCI_OBJ']:
+                        datas.append(tmp_data)
+                        headers.append(tmp_header)
+                else:
+                    datas.append(tmp_data)
+                    headers.append(tmp_header)
+            else:
+                warn_user("File "+tmp_file+" does not look like a SUVI L1b file. Skipping.")
+
+    if len(datas) == 1 and len(headers) == 1:
+        # Make a single map if it is just one file
+        suvimap = sunpy.map.Map(datas[0], headers[0])
+        return suvimap
+    elif len(datas) > 1 and len(headers) > 1:
+        # Make a map sequence for multiple files
+        suvimap = sunpy.map.Map(list(zip(datas, headers)), sequence=True)
+        return suvimap
+    else:
+        warn_user("List of data/headers is empty.")
+        return None
+
+
+def fix_L1b_header(input_filename):
+    """
+    Fix a SUVI L1b FITS file header (broken due to the wrong
+    implementation of the CONTINUE keyword convention).
```

Comment:
What is the problem here? If this is a bug in astropy, we can raise it there and have them fix it or explain to us what is going on.

Reply:
This is a longer story: the current SUVI L1b FITS files that can be downloaded and used from NOAA/with fido are not being created by us. Unfortunately, the creator has implemented the CONTINUE convention incorrectly (two blanks missing after CONTINUE). We have complained about this for years. It has not been fixed yet, and I don't know if it ever will, or if the data will ever be reprocessed so that this is not a problem anymore. A while back, with an astropy version <4.2.1, I wrote this function based on astropy's .to_string() method in the Header class to clean up the header, so that when you use read_suvi, this would all get fixed silently in the background. With astropy 4.2.1, they introduced a header consistency check which will not allow me to use the .to_string() method anymore, it just quits and complains the that the headers are not compliant with the FITS standard. I have opened an [issue](https://github.com/astropy/astropy/issues/11815) about this, but there seems to be little interest in getting this fixed. Maybe if you chime in on the issue, your word will carry more weight than mine. I was just getting tired of waiting, hence the alternative approach for astropy >=4.2.1.

Reply:
Thanks for explaining that, I will see if I can ping someone on astropy about that. If not, maybe I can try to patch the to_string method.